### PR TITLE
User Interface: create cluster modal and form

### PR DESCRIPTION
**Pull request by eap**
Add a cluster-create button that launches a modal. The modal runs simple name validation and builds the request. Additionally, the modal will briefly show a progress spinner while waiting for the API to return a successful response. On success the modal notifies the application state container to refresh the entire list.

See a screenshot of the modal [here](https://docs.google.com/document/d/16sEFuGvr4omhDiB1iVxc1ogeYkb1zIUL-6txKew8oJ4)

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: ui/src/components/CreateClusterForm.js [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+const DEFAULT_WORKER_MACHINE_TYPE = "n1-standard-2";
+const DEFAULT_WORKER_DISK_SIZE = 200;
+
+// Regex for GCP Project names.
+const projectRegex = /^[a-z][a-z0-9_-]{0,50}[a-z0-9]$/;
+
+// Regex for cluster names.
+const clusterRegex = /^[a-z][a-z0-9-]{0,25}[a-z0-9]$/;
+
+const validMachineTypes = [
```
Comment (rtitle) [inline ui/src/components/CreateClusterForm.js]: Why limit it to this set? What if someone needs a highmem machine type?

File: ui/src/components/CreateClusterForm.js [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+      requestInProgress: true,
+    });
+
+    var createRequest = {
+      labels: {},
+      machineConfig: {
+        // Worker config.
+        numberOfWorkers: this.state.numberOfWorkers,
+        workerMachineType: this.state.workerMachineType,
+        numberOfPreemptibleWorkers: 0,
```
Comment (rtitle) [inline ui/src/components/CreateClusterForm.js]: Why disallow preemptibles? In my experience people like to use them as they are _much_ cheaper.

Comment (eap) [reply ui/src/components/CreateClusterForm.js]: I agree with the sentiment, see my comment on this PR.

File: ui/src/components/CreateClusterForm.js (hunk 1 above)
Comment (eap) [reply ui/src/components/CreateClusterForm.js]: Added a few more types

File: ui/src/components/CreateClusterForm.js [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+
+  createClusterRequest = (event) => {
+    // Sets state to disable submission & and show loading spinner.
+    this.setState({
+      createRequestValid: false,
+      requestInProgress: true,
+    });
+
+    var createRequest = {
+      labels: {},
```
[the remark this answers is not in the document]
Comment (eap) [reply ui/src/components/CreateClusterForm.js]: Agreed.

**Comment by rtitle** [comment]
OK thanks, I can merge this then.

**Pull request merged**